bin/eslint.js

```diff
@@ -45,33 +45,74 @@ function buildIssueJson(message, path) {
   return JSON.stringify(issue);
 }
 
-// Uses glob to traverse code directory and find files to analyze,
-// excluding files passed in with by CLI config, and including only
-// files in the list of desired extensions
-function fileWalk(excludePaths, extensions){
-  var analysisFiles = [];
-  var allFiles = glob.sync("/code/**/**", {});
-
-  allFiles.forEach(function(file, i, a){
-    if(excludePaths.indexOf(file.split("/code/")[1]) < 0) {
-      if(!fs.lstatSync(file).isDirectory()) {
-        var extension = "." + file.split(".").pop();
-
-        if(extensions.indexOf(extension) >= 0) {
-          analysisFiles.push(file);
+function isFileWithMatchingExtension(file, extensions) {
+  var stats = fs.lstatSync(file);
+  var extension = "." + file.split(".").pop();
```

Comment:
This line returns the last fragment in the string, i.e. `.gz` or `.slim`. So I think this is what we want. Even in a case where `js` is in the middle of the string, such as `application.js.erb`, we wouldn't to consider that analyzable by ESLint.